Add horizontal resize of ai chat panel

FadhlanR opened the pull request:
Ticket: CS-6430


https://github.com/cardstack/boxel/assets/12637010/9a469b2a-cd7b-4502-adbd-cf0362f557c2

FadhlanR committed:
Add horizontal resize handle on ai chat panel

FadhlanR committed:
lint fix

FadhlanR committed:
Add resize handle to code submode

backspace commented:
I’m not sure of the boundaries of this and other issues, maybe this is 6485, but I noticed I’m unable to scroll the messages container and when I tab to the field to add another message everything shifts upward:

![screencast 2024-01-18 12-34-16](https://github.com/cardstack/boxel/assets/43280/081c6625-947e-45de-b29b-296d0f74019b)

The proximity detection for showing and highlighting the handle is great, I found the snapping at the minimal width a bit jarring, unclear how that could be improved though

![screencast 2024-01-18 12-41-40](https://github.com/cardstack/boxel/assets/43280/402653b9-9c22-499f-a9cc-e9719a9dc93d)

backspace commented:
I also meant to ask: I’d expect the open/closed state of the panel to be persistent across refreshes, as well as its width. I don’t see an open issue for this and realise it’s not in the issue as defined though!

FadhlanR committed:
Fix scroll issue and collapsible panel

FadhlanR commented:
@backspace , Thank you for the feedback. I've addressed the scroll issue in the AI Assistant panel and added a `collapsible` argument to `ResizablePanel`. By default, this argument is set to true, but if it's set to false, the ResizablePanel won't be collapsed.


https://github.com/cardstack/boxel/assets/12637010/88f5ab38-8474-425a-ba69-1a0c51f9587e

FadhlanR commented:
> I also meant to ask: I’d expect the open/closed state of the panel to be persistent across refreshes, as well as its width. I don’t see an open issue for this and realise it’s not in the issue as defined though!

@backspace do you think it is better to make the open/closed and width state persistent in this PR?

FadhlanR committed:
Fix boxel UI test

backspace commented:
> do you think it is better to make the open/closed and width state persistent in this PR?

I’m uncertain! Let’s just say leave it for now, something to consider for the future.

The scrolling and non-collapse look good, thanks!

ETA something seems to be terribly wrong with the tests and therefore Percy, let us know when it’s ready again

FadhlanR committed:
Fix host tests

FadhlanR committed:
Fix host test

FadhlanR commented:
I've fixed boxel ui test and host test. This PR is ready for another review. cc: @backspace

backspace commented on packages/boxel-ui/addon/src/components/resizable-panel-group/index.gts:
```diff
@@ -177,13 +224,22 @@ export default class ResizablePanelGroup extends Component<Signature> {
   }
 
+  @action
+  unRegisterPanel(id: number) {
```
This is not a big deal but `unregister` is fine, no need for caps

backspace reviewed (approved):
looking good, I appreciate all the refinements 🎉

jurgenwerk commented:
I think there should be a limit  how much you can drag this to the left. This happens if you drag it all the way to the left:

<img width="477" alt="image" src="https://github.com/cardstack/boxel/assets/273660/4d867c92-e2bd-4736-9228-8b162e4fbdfb">

lukemelia commented:
Is the ResizeHandler component the thing responsible for rendering the handle? If so, should it be called ResizeHandle instead of ResizeHandler? I usually think of a "handler" as something that responds to events being triggered.

FadhlanR committed:
Add iteract mode min length and rename ResizeHandler to ResizeHandle

FadhlanR committed:
Rename function

FadhlanR committed:
Merge branch 'main' into cs-6430-horizontal-resize-of-ai-chat-panel

FadhlanR commented:
@jurgenwerk I've added minimum length and made `collapsibale` to `false` for stack items panel. So we cannot make the stack items panel disappear.


https://github.com/cardstack/boxel/assets/12637010/fc75420a-d067-4d17-8412-e11745f5c1de

FadhlanR committed:
Merge branch 'main' into cs-6430-horizontal-resize-of-ai-chat-panel
